clients/src/test/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtilsTest.java

```diff
@@ -111,5 +116,20 @@ public void testValidateIntervalMsInvalid(int pushIntervalMs) {
     public void testPreferredCompressionType() {
         assertEquals(CompressionType.NONE, ClientTelemetryUtils.preferredCompressionType(Collections.emptyList()));
         assertEquals(CompressionType.NONE, ClientTelemetryUtils.preferredCompressionType(null));
+        assertEquals(CompressionType.NONE, ClientTelemetryUtils.preferredCompressionType(Arrays.asList(CompressionType.NONE, CompressionType.GZIP)));
+        assertEquals(CompressionType.GZIP, ClientTelemetryUtils.preferredCompressionType(Arrays.asList(CompressionType.GZIP, CompressionType.NONE)));
+    }
+
+    @ParameterizedTest
+    @EnumSource(CompressionType.class)
+    public void testCompressDecompress(CompressionType compressionType) {
+        byte[] testString = "test string".getBytes(StandardCharsets.UTF_8);
+        ByteBuffer compressed = ClientTelemetryUtils.compress(testString, compressionType);
+        assertNotNull(compressed);
```

Comment:
Should we check if array length was reduced?

Reply:
Done.